nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java

```diff
@@ -406,9 +406,34 @@ public static INDArray expandDims(INDArray input, int dimension) {
         long[] indexes = new long[input.rank() + 1];
         for (int i = 0; i < indexes.length; i++)
             indexes[i] = i < dimension ? shape[i] : i == dimension ? 1 : shape[i - 1];
-        return input.reshape(input.ordering(), indexes);
+        return input.reshape(indexes);
```

Comment:
The order arg here is quite important given the way ND4J works with reshaped arrays being a view of the original.